connector/dialect.tdd

```diff
@@ -1,541 +1,326 @@
+    <function group='cast' name='DECIMAL' return-type='real'>
+      <formula>CAST(%1 AS DECIMAL)</formula>
       <argument type='real' />
     </function>
-    <function group='cast' name='DATE' return-type='date'>
-      <formula>CAST(%1 AS DATE)</formula>
-      <argument type='str' />
-    </function>
-    <function group='cast' name='DATE' return-type='date'>
-      <formula>CAST(%1 AS DATE)</formula>
-      <argument type='datetime' />
+    <function group='cast' name='DECIMAL' return-type='real'>
+      <formula>CAST(%1 AS DECIMAL)</formula>
+      <argument type='int' />
     </function>
-    <function group='cast' name='DATETIME' return-type='datetime'>
-      <formula>(CAST(&apos;1900-01-01 00:00:00&apos; AS TIMESTAMP) + %1 * INTERVAL &apos;1 DAY&apos;)</formula>
-      <argument type='real' />
+    <function group='cast' name='DECIMAL' return-type='real'>
+      <formula>CAST(%1 AS DECIMAL)</formula>
+      <argument type='bool' />
     </function>
-    <function group='cast' name='DATETIME' return-type='datetime'>
-      <formula>CAST(%1 AS TIMESTAMP)</formula>
+    <function group='cast' name='FLOAT' return-type='real'>
+      <formula>CAST(%1 AS DOUBLE)</formula>
       <argument type='str' />
     </function>
-    <function group='cast' name='DATETIME' return-type='datetime'>
-      <formula>CAST(%1 AS TIMESTAMP)</formula>
-      <argument type='datetime' />
-    </function>
     <function group='cast' name='FLOAT' return-type='real'>
-      <formula>(CASE
-	WHEN %1 THEN 1.0
-	WHEN NOT %1 THEN 0.0
-	ELSE NULL END)</formula>
-      <argument type='bool' />
+      <formula>CAST(%1 AS DOUBLE)</formula>
+      <argument type='datetime' />
     </function>
     <function group='cast' name='FLOAT' return-type='real'>
-      <formula>CAST(%1 AS DOUBLE PRECISION)</formula>
+      <formula>(%1)</formula>
       <argument type='real' />
     </function>
     <function group='cast' name='FLOAT' return-type='real'>
-      <formula>CAST(%1 AS DOUBLE PRECISION)</formula>
-      <argument type='str' />
+      <formula>CAST(%1 AS DOUBLE)</formula>
+      <argument type='int' />
     </function>
     <function group='cast' name='FLOAT' return-type='real'>
-      <formula>((EXTRACT(EPOCH FROM %1) / ( 60 * 60 * 24)) + (365 * 70 + 17))</formula>
-      <argument type='datetime' />
+      <formula>CAST(%1 AS DOUBLE)</formula>
+      <argument type='bool' />
     </function>
     <function group='cast' name='INT' return-type='int'>
-      <formula>(CASE
-	WHEN %1 THEN 1
-	WHEN NOT %1 THEN 0
-	ELSE NULL END)</formula>
-      <argument type='bool' />
+      <formula>CAST(%1 AS INT)</formula>
+      <argument type='str' />
     </function>
     <function group='cast' name='INT' return-type='int'>
-      <formula>CAST(TRUNC(%1) AS BIGINT)</formula>
-      <argument type='real' />
+      <formula>CAST(%1 AS INT)</formula>
+      <argument type='datetime' />
```

Comment:
Based on this PR, it seems Tableau's `int` covers 32- and 64-bit ints. In MongoSQL/MongoDB these are separate types, `int` and `long`, respectively. We don't support converting a bson `date` (Tableau `datetime`) to a 32-bit `int` or converting a 32-bit `int` to a bson `date`.

We should remove this version of the `INT` CAST function. I'm not sure we should/could remove the `DATETIME` CAST function whose argument type is `int` since that could be covering the `long` case... Regardless, let's get rid of this version of `INT`.

Reply:
Thanks for that info, I've removed the `INT` CAST of argument type='datetime'.